src/lib/actions/AbstractRunAction.ts

```diff
@@ -80,6 +81,7 @@ export abstract class AbstractRunAction implements Action {
 	}
 
 	async run(inputs: Inputs): Promise<AnyJson> {
+		Controller.setActivePmdCommandInfo(inputs['preview-pmd7'] ? new Pmd7CommandInfo() : new Pmd6CommandInfo());
```

Comment:
For sanity sake... and since our testing framework could run many invocations of run... I think it is best to always set the state instead of trying to clean up after ourselves (even for PMD6). As soon as we do a full refactoring and remove global state from our code - then we won't need to worry so much.